apps/frontend/src/components/xpress/XpressProposalTable.tsx

```diff
@@ -245,6 +329,136 @@ const XpressProposalTable = () => {
     },
   ];
 
+  const statusColumn = () => [
+    {
+      title: 'Status',
+      field: 'statusName',
+      sorting: false,
+      render: (rowData: ProposalViewData) => {
+        if (loadingProposalStatuses) {
+          return <UOLoader style={{ marginLeft: '50%', marginTop: '20px' }} />;
+        }
+
+        const fieldValue = proposalStatuses.find(
+          (ps) => ps.name === rowData.statusName
+        );
+
+        // Disallow setting submitted or draft status, unless user officer
+        let availableStatuses;
+        if (isUserOfficer) {
+          availableStatuses = xpressStatuses;
+        } else {
+          availableStatuses = xpressStatuses.filter(
+            (status) =>
+              status.shortCode !== StatusCode.SUBMITTED_LOCKED &&
+              status.shortCode !== StatusCode.DRAFT &&
+              status.shortCode !== StatusCode.EXPIRED
+          );
+        }
+
+        xpressStatuses.sort((a, b) => {
+          return (
+            xpressStatusCodes.indexOf(a.shortCode as StatusCode) -
+            xpressStatusCodes.indexOf(b.shortCode as StatusCode)
+          );
+        });
+
+        // Always show the current status at the top of the dropdown
+        if (fieldValue) {
+          const currentIndex = availableStatuses.findIndex(
+            (status) => status.id === fieldValue.id
+          );
+          if (currentIndex > -1) {
+            availableStatuses.splice(currentIndex, 1);
+          }
+          availableStatuses.unshift(fieldValue);
+        }
+
+        const isInstrumentAbsent = (rowData.instruments?.length ?? 0) === 0;
+        const isStatusDraft = fieldValue?.shortCode === StatusCode.DRAFT;
+        const isStatusSubmitted =
+          fieldValue?.shortCode === StatusCode.SUBMITTED_LOCKED;
+        const isStatusUnsuccessful =
+          fieldValue?.shortCode === StatusCode.UNSUCCESSFUL;
+        const isStatusApproved = fieldValue?.shortCode === StatusCode.APPROVED;
+        const isStatusFinished = fieldValue?.shortCode === StatusCode.FINISHED;
+        const isStatusExpired = fieldValue?.shortCode === StatusCode.EXPIRED;
```

Comment:
If all of this is related to a status would we not benefit from putting then into one object.

Reply:
I've put the frontend statuses in an object but I left the backend as I think it negatively affected readability considering we have two types of statuses there already (current and new)